app/utils/utils.ts

```diff
@@ -144,3 +145,20 @@ export function useUpdateSearchParams(): [
   };
   return [searchParams, updateSearchParams, setSearchParams];
 }
+
+export function readToggleOn(
+  searchParams: URLSearchParams,
+  key: ToggleParamKey,
+): boolean {
+  return searchParams.get(key) !== "off";
+}
+
+export function useToggleParam(
```

Comment:
I'm not totally convinced these helper functions and the corresponding type sin `types.ts` are totally justified. What do they offer beyond just using `useUpdateSearchParams` directly?